### PR TITLE
#11133: Improve various things about the wheel, including removal of `patchelf` and linking runtime assets to cwd

### PR DESCRIPTION
**Pull request by tt-rkim**
### Ticket

#11133

### Problem description
Provide context for the problem.

### What's changed
Describe the approach used to solve the problem.
Summarize the changes made and its impact.

### Checklist
- [ ] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: .github/workflows/eager-package.yaml [hunk 1]
```diff
@@ -18,7 +18,6 @@ jobs:
     runs-on: ${{ inputs.os }}
     environment: ${{ inputs.environment }}
     env:
-      TT_METAL_ENV: ${{ vars.TT_METAL_ENV }}
       ARCH_NAME: ${{ inputs.arch }}
       TT_METAL_CREATE_STATIC_LIB: 1
```
Comment (tt-rkim) [inline .github/workflows/eager-package.yaml:22]: must remove

File: setup.py [hunk 2]
```diff
@@ -95,41 +94,41 @@ def get_build_env():
...
+
+        # Currently, the ttnn (ttnn/_ttnn.so)
+        # both link to the tt_metal runtime. The specific thing in
+        # ttnn linking to tt_metal is likely the implementation of
+        # ttnn.manage_device.
+        # However, because of the singleton design of
+        # tt_cluster in tt_metal, ttnn will have a
+        # copy of the cluster object, causing a hang during
+        # device operations in ttnn, such as calling
+        # output = ttnn.to_torch(output).
```
Comment (ayerofieiev-tt) [inline setup.py:127]: We will not statically build = We will not build as a static library?

at the same time we say shared libs off in next line of code

File: setup.py [hunk 3]
```diff
@@ -95,41 +94,41 @@ def get_build_env():
...
-        build_dir = Path(os.environ.get("TT_METAL_HOME") + "/build")
-        if not build_dir.exists():
-            build_dir.mkdir(parents=True)
-
-        cmake_args = []
+        source_dir = CMakeBuild.get_working_dir()
+        build_dir = source_dir / "build"
+
+        # Currently, the ttnn (ttnn/_ttnn.so)
+        # both link to the tt_metal runtime. The specific thing in
```
[the remark this answers is not in the document]
Comment (tt-rkim) [reply setup.py:120]: Honestly with how we're packaging the SOs... I'm wondering if we need this at all. Let me try that out!

**Comment by tt-rkim** [comment]
Will add much more documentation + explanation of what's going on per @ayerofieiev-tt 's suggestion

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by tt-rkim** [comment]
notes for merging:

- `ARCH_NAME` / `TT_METAL_HOME` exclusion and changing workflows for that
- ensuring we're in wheel mode

**Pull request merged**